docs/spec/dns-integration.md

```diff
@@ -0,0 +1,89 @@
+DNS Integration for NSM
+============================
+
+Specification
+-------------
+
+Network Service Mesh needs to be able to provide a workload with DNS service from Network Services Without breaking K8s DNS
+
+Implementation details (optional)
+---------------------------------
+
+####nsm-coredns
+`nsm-corends` is a docker image based on [coredns](https://github.com/coredns/coredns.io/blob/master/content/manual/what.md). The difference with the original `coredns` in the set of plug-ins. 
+The image uses only next `coredns` plugins:
+* `bind`
+* `hosts`
+* `log`
+* `reload`
+
+Also, it includes special custom plugin `fanout` (see below).	
+#### Fanout plugin
+`fanout` is custom [plugin for coredns](https://coredns.io/manual/plugins/).
+The fanout plugin re-uses already opened sockets to the upstreams. It supports TCP and DNS-over-TLS and uses in-band health checking. 
+For each incoming DNS query that hits the CoreDNS fanout plugin, it will be replicated in parallel to each listed IP. The first non-negative response from any of the queried DNS Servers will be forwarded as a response to the request.
+
+#### How use nsm-coredns as default name server for pod?
+1) Deploy corefile configmap 
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns
+  namespace: kube-system
+data:
+  Corefile: |
+    .:53 {
+        errors
+        health
+        ready
+        kubernetes CLUSTER_DOMAIN REVERSE_CIDRS {
+          pods insecure
+          fallthrough in-addr.arpa ip6.arpa
+        }FEDERATIONS
+        prometheus :9153
+        forward . UPSTREAMNAMESERVER
+        cache 30
+        loop
+        reload
+        loadbalance
+    }STUBDOMAINS
+```
+2) Deploy nsm-coredns. You can deploy it as sidecar for your pod (see below) or as separate pod.
```

Comment:
How would nsm-coredns work in a separate Pod given that it may need to make DNS queries over vWires that are only available in the consuming Pod, not a secondary Pod?

Reply:
good catch. I simply meant that we can deploy `nsm-coredns` as separate pod and use it's IP in another pod as `Nameserver`. But I think it can confuse the user. Finally, doc updated.